### PR TITLE
4616: Validation

### PR DESCRIPTION
**Pull request by kat1906**
Story: [AB#4616](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4616)
Task: [AB#7091](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7091)

### CODE REVIEW

File: app/pages/sections/supplier/search/contextCreator.js [hunk 1]
```diff
@@ -11,15 +11,7 @@ export const getContext = ({ orderId }) => {
...
-    ...modifiedManifest,
-    errors: allErrors,
-    title: `${manifest.title} ${orderId}`,
-    backlinkHref: `${baseUrl}/organisation/${orderId}`,
-  };
-};
+export const getErrorContext = params => ({
+  ...getSectionErrorContext({ ...params, manifest }),
+  title: `${manifest.title} ${params.orderId}`,
+});
```
Comment (alibombaye) [inline app/pages/sections/supplier/search/contextCreator.js:17]: oooh I see!!! I was wondering how you was going to do this. So obvious 😂

File: app/routes.test.js [hunk 2]
```diff
@@ -371,8 +376,10 @@ describe('routes', () => {
         .send({ _csrf: csrfToken })
         .expect(200)
         .then((res) => {
-          expect(res.text.includes('error with put call for call-off-ordering-party')).toEqual(true);
+          expect(res.text.includes('data-test-id="call-off-ordering-party-page"')).toEqual(true);
+          expect(res.text.includes('data-test-id="error-summary"')).toEqual(true);
           expect(res.text.includes('data-test-id="error-title"')).toEqual(false);
+          descriptionController.getDescriptionErrorContext.mockReset();
```
Comment (alibombaye) [inline app/routes.test.js:382]: think we can remove this line

File: app/pages/sections/call-off-ordering-party/ui.test.js [hunk 3]
```diff
@@ -247,3 +256,67 @@ test('should navigate to task list page if save button is clicked and data is va
...
+
+test('should show text fields as errors with error message when there are validation errors', async (t) => {
+  nock(orderApiUrl)
+    .put('/api/v1/orders/order-id/sections/ordering-party')
+    .reply(400, putOrderingPartyErrorResponse);
+
+  await pageSetup(t, true);
+  await t.navigateTo(pageUrl);
+
+  const descriptionPage = Selector('[data-test-id="call-off-ordering-party-page"]');
```
Comment (alibombaye) [inline app/pages/sections/call-off-ordering-party/ui.test.js:289]: rename this const. `callOffOrderingPartyPage`?

File: app/pages/sections/call-off-ordering-party/ui.test.js [hunk 4]
```diff
@@ -247,3 +256,67 @@ test('should navigate to task list page if save button is clicked and data is va
...
+  const saveButton = Selector('[data-test-id="save-button"] button');
+  const descriptionField = descriptionPage.find('[data-test-id="question-telephoneNumber"]');
```
Comment (alibombaye) [inline app/pages/sections/call-off-ordering-party/ui.test.js:291]: rename this const `telephoneNumberField`?

File: app/pages/sections/call-off-ordering-party/ui.test.js [hunk 5]
```diff
@@ -247,3 +256,67 @@ test('should navigate to task list page if save button is clicked and data is va
...
+
+  await t
+    .expect(descriptionField.find('[data-test-id="textarea-field-error"]').exists).notOk()
```
Comment (alibombaye) [inline app/pages/sections/call-off-ordering-party/ui.test.js:294]: this should be a textfield that is should be trying to find right?

File: app/pages/sections/call-off-ordering-party/ui.test.js [hunk 6]
```diff
@@ -247,3 +256,67 @@ test('should navigate to task list page if save button is clicked and data is va
...
+  await t
+    .expect(errorSummary.exists).notOk()
+    .click(saveButton);
+
+  await t
+    .expect(errorSummary.exists).ok()
+
+    .click(errorSummary.find('li a').nth(0))
+    .expect(getLocation()).eql(`${pageUrl}#telephoneNumber`);
+});
```
Comment (alibombaye) [inline app/pages/sections/call-off-ordering-party/ui.test.js:322]: In the past we have ui-tested all the fields. Is this worth doing here also?

File: app/routes.test.js (hunk 2 above)
Comment (kat1906) [reply app/routes.test.js:382]: forgot to change it

File: app/pages/sections/call-off-ordering-party/ui.test.js (hunk 6 above)
Comment (kat1906) [reply app/pages/sections/call-off-ordering-party/ui.test.js:322]: yep

File: app/pages/sections/call-off-ordering-party/ui.test.js (hunk 5 above)
Comment (kat1906) [reply app/pages/sections/call-off-ordering-party/ui.test.js:294]: good spot

**Pull request merged**